### PR TITLE
Inline toolbar tabs based off selected style

### LINKED ISSUE #493: Use inline toolbar style on tab bar
Since elementary/stylesheet#379 was merged. This will require:

- [ ] Defining `@define-color tab_base_color $FOO` in application CSS where `$FOO` is the background color of the current syntax scheme. 
- [ ] Changing that variable when the style scheme changes. We _could_ try to introspect into the style scheme to get that color, _or_ we could just define three versions based on the style scheme switcher, since we know what scheme we're going to use there. Just depends on how complex we want to get.
- [ ] Adding the `.inline-toolbar` style class to the Dynamic Notebook widget.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/62180436-use-inline-toolbar-style-on-tab-bar?utm_campaign=plugin&utm_content=tracker%2F61917289&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F61917289&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by davidmhewitt**
Fixes #493

### CODE REVIEW

File: src/Widgets/DocumentView.vala [hunk 1]
```diff
@@ -84,9 +86,43 @@ public class Scratch.Widgets.DocumentView : Granite.Widgets.DynamicNotebook {
...
+            // Default gtksourceview background color is white
+            var color = "#FFFFFF";
+            if (text_color_data != null) {
+                // If the current style has a background color, use that
+                color = text_color_data.background;
+            }
+
+            var define = "@define-color tab_base_color %s;".printf (color);
+            style_provider.load_from_data (define);
+            style_context.add_class ("inline-toolbar");
```
Comment (danrabbit) [inline src/Widgets/DocumentView.vala]: This should be `Gtk.STYLE_CLASS_INLINE_TOOLBAR`

File: src/Widgets/DocumentView.vala [hunk 2]
```diff
@@ -84,9 +86,43 @@ public class Scratch.Widgets.DocumentView : Granite.Widgets.DynamicNotebook {
...
 
+    private void update_inline_tab_colors () {
+        var sssm = Gtk.SourceStyleSchemeManager.get_default ();
+        var style_context = get_style_context ();
+
+        if (settings.style_scheme in sssm.scheme_ids) {
+            var theme = sssm.get_scheme (settings.style_scheme);
+            var text_color_data = theme.get_style ("text");
+
+            // Default gtksourceview background color is white
```
Comment (cassidyjames) [inline src/Widgets/DocumentView.vala]: I know the default color that inline notebooks pick up is white. Do we still want/need to define this here?

File: src/Widgets/DocumentView.vala [hunk 3]
```diff
@@ -84,9 +86,47 @@ public class Scratch.Widgets.DocumentView : Granite.Widgets.DynamicNotebook {
...
+            try {
+                style_provider.load_from_data (define);
+                return;
+            } catch (Error e) {
+                critical ("Unable to set inline tab styling, going back to classic notebook tabs");
+            }
+        }
+
+        // Fallback to a non inline toolbar if something went wrong above
+        style_context.remove_class ("inline-toolbar");
```
Comment (danrabbit) [inline src/Widgets/DocumentView.vala]: This one too ;D

File: src/Widgets/DocumentView.vala (hunk 2 above)
Comment (cassidyjames) [reply src/Widgets/DocumentView.vala]: @davidmhewitt makes sense, thanks for the explanation!

**Pull request merged**